jfrog-cli/artifactory/utils/golang/project/dependencies/dependenciesutils.go

```diff
@@ -0,0 +1,452 @@
+	// Remove go.sum file to avoid checksum conflicts with the old go.sum
+	goSum := filepath.Join(filepath.Dir(path), "go.sum")
+	exists, err := fileutils.IsFileExists(goSum, false)
+	if err != nil {
+		return
+	}
+
+	if exists {
+		err = os.Remove(goSum)
+		if errorutils.CheckError(err) != nil {
+			return
+		}
+	}
+
+	if shouldRunGoModCommand {
+		// Running go mod tidy command
+		err = golang.RunGoModTidy()
+		if err != nil {
+			return
+		}
+	}
+	if shouldRunGoGraph {
+		// Running go mod graph command
+		output, err = golang.GetDependenciesGraph()
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+// Downloads the mod file from Artifactory to the Go cache
+func downloadModFileFromArtifactoryToLocalCache(cachePath, targetRepo, name, version string, details *config.ArtifactoryDetails, client *httpclient.HttpClient) string {
+	pathToModuleCache := filepath.Join(cachePath, name, "@v")
+	dirExists, err := fileutils.IsDirExists(pathToModuleCache, false)
+	if err != nil {
+		log.Debug("Received and error:", err)
+		return ""
+	}
+
+	if dirExists {
+		url := details.Url + "api/go/" + targetRepo + "/" + name + "/@v/" + version + ".mod"
+		log.Debug("Downloading mod file from Artifactory:", url)
+		auth, err := details.CreateArtAuthConfig()
+		if err != nil {
+			log.Debug("Received and error:", err)
+			return ""
+		}
+		downloadFileDetails := &httpclient.DownloadFileDetails{
+			FileName: version + ".mod",
+			// Artifactory URL
+			DownloadPath:  url,
+			LocalPath:     pathToModuleCache,
+			LocalFileName: version + ".mod",
+		}
+		resp, err := client.DownloadFile(downloadFileDetails, "", auth.CreateHttpClientDetails(), 3, false)
+		if err != nil {
+			log.Debug("Received and error:", err)
+			return ""
+		}
+
+		log.Debug(fmt.Sprintf("Received %d from Artifactory %s", resp.StatusCode, url))
+		return filepath.Join(downloadFileDetails.LocalPath, downloadFileDetails.LocalFileName)
+	}
+	return ""
+}
+
+func GetRegex() (regExp *RegExp, err error) {
+	emptyRegex, err := utils.GetRegExp(`^\s*require (?:[\(\w\.@:%_\+-.~#?&]?.+)`)
+	if err != nil {
+		return
+	}
+
+	indirectRegex, err := utils.GetRegExp(`(// indirect)$`)
+	if err != nil {
+		return
+	}
+
+	regExp = &RegExp{
+		notEmptyModRegex: emptyRegex,
+		indirectRegex:    indirectRegex,
+	}
+	return
+}
+
+func downloadAndCreateDependency(cachePath, name, version, fullDependencyName, targetRepo string, downloadedFromArtifactory bool, details *config.ArtifactoryDetails) (*Package, error) {
+	// Dependency is missing within the cache. Need to download it...
+	err := downloadDependency(downloadedFromArtifactory, fullDependencyName, targetRepo, details)
+	if err != nil {
+		return nil, err
+	}
+	// Now that this dependency in the cache, get the dependency object
+	dep, err := createDependency(cachePath, name, version)
+	if err != nil {
+		return nil, err
+	}
+	return dep, nil
+}
+
+func logErrorIfOccurred(err error) {
+	if err != nil {
+		log.Debug("Received and error:", err)
```

Comment:
Type in the log message.
I suggest changing this to log.Error

Reply:
Done